H5MD: Refactoring

hirschsn opened the pull request:
Changes:
- Simplification of file opening/creation
- Masses, types can be written time dependant
- Unification of dimensionalities
- Change of interface (use enum: File.write(W_POS | W_F | W_V), etc.)
- Change of python interface: file.write("pfv"')... one char per output field. Somehow resembles the old tcl interface; we could also use a list or whatever)

Please review this patch thoroughly. The test script produces nice output. I don't particularly like the python interface; we could also just make the constants public (W_POS, W_F, ...).

KaiSzuttor commented:
I get the following error when running the sample:
Error: "The MPI_Comm_free() function was called after MPI_FINALIZE was invoked"
This might be due to unclosed hdf5 objects. I'll have a look.

KaiSzuttor commented:
Regarding the interface, we should not put any effort in making a nice one, because it will change when we switch to the script_interface approach.

hirschsn commented:
You're right. I just introduced the change of interface in this commit since I removed the old write(bool, bool, bool) in the C++ code. But in the pull request message I meant we should think about the final interface.
